sdk/core/Azure.Core/src/ResourceIdentifier.cs

```diff
@@ -0,0 +1,429 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// An Azure Resource Manager resource identifier.
+    /// </summary>
+    public sealed class ResourceIdentifier : IEquatable<ResourceIdentifier>, IComparable<ResourceIdentifier>
+    {
+        private const char Separator = '/';
+
+        private static ResourceType TenantResourceType = new ResourceType("Microsoft.Resources/tenants");
+        private static ResourceType SubscriptionResourceType = new ResourceType("Microsoft.Resources/subscriptions");
+        private static ResourceType ResourceGroupResourceType = new ResourceType("Microsoft.Resources/resourceGroups");
+        private static ResourceType ProviderResourceType = new ResourceType("Microsoft.Resources/providers");
```

Comment:
It looks like we would only need to create an instance of one of these to instantiate a valid ResourceIdentifier -- the others are used for comparisons to determine the resource type and we could use string comparisons there instead.  If my understanding is correct, would you mind refactoring here to reduce allocations?